internal/identity/github.go

```diff
@@ -0,0 +1,171 @@
+package identity
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"gopkg.in/square/go-jose.v2/jwt"
+
+	"github.com/pomerium/pomerium/internal/httputil"
+	"github.com/pomerium/pomerium/internal/sessions"
+	"github.com/pomerium/pomerium/internal/version"
+
+	"golang.org/x/oauth2"
+)
+
+const (
+	defaultGitHubProviderURL = "https://github.com"
+	githubAuthURL            = "/login/oauth/authorize"
+	githubUserURL            = "https://api.github.com/user"
+	githubUserGroupURL       = "https://api.github.com/user/orgs"
+	githubRevokeURL          = "https://github.com/oauth/revoke"
+
+	refreshDeadline = time.Minute * 60
```

Comment:
maybe add a comment explaining why we have a refresh deadline here (oauth2 provider, not oidc etc).

Reply:
Cool, that'll make sense. thanks